File: test-fixtures/configs/nodejs/src/hooks.js
/**
 * Hook implementations for Test Node.js CLI
 * 
 * This file contains the business logic for your CLI commands.
 * Implement the hook functions below to handle your CLI commands.
 * 
 * IMPORTANT: Hook names must use snake_case with 'on_' prefix
 * Example:
 * - Command 'hello' -> Hook function 'on_hello'
 * - Command 'hello-world' -> Hook function 'on_hello_world'
 */

// Import any modules you need here
import fs from 'fs';
import path from 'path';
/**
 * Say hello * @returns {Promise<number>} - Exit code (0 for success)
 */
async function on_hello() {    // Generic command implementation
    console.log('Executing hello command...');    console.log('✅ Command completed successfully!');    
    return 0;
}
/**
 * Start server * @returns {Promise<number>} - Exit code (0 for success)
 */
async function on_serve() {    // Generic command implementation
    console.log('Executing serve command...');    console.log('✅ Command completed successfully!');    
    return 0;
}
/**
 * Build a project * @returns {Promise<number>} - Exit code (0 for success)
 */
async function on_build_project() {
    // Add your business logic here
    console.log('Hook on_build_project called');    
    // Return 0 for success, non-zero for error
    return 0;
}
// Export all hook functions
export {    on_hello,    on_serve,    on_build_project,};

File: test-fixtures/configs/rust/src/hooks.rs
// Hook implementations for test Rust CLI

use clap::ArgMatches;
use anyhow::Result;

pub fn on_hello(_matches: &ArgMatches) -> Result<()> {
    println!("Hello from Rust CLI!");
    Ok(())
}

pub fn on_build(_matches: &ArgMatches) -> Result<()> {
    println!("Build command executed!");
    Ok(())
}

pub fn on_build_project(_matches: &ArgMatches) -> Result<()> {
    println!("Building project...");
    Ok(())
}

pub fn on_serve(_matches: &ArgMatches) -> Result<()> {
    println!("Starting server...");
    Ok(())
}